src/main/java/com/pusher/client/channel/impl/PrivateEncryptedChannelImpl.java

```diff
@@ -45,22 +48,39 @@ public void bind(final String eventName, final SubscriptionEventListener listene
         super.bind(eventName, listener);
     }
 
-    private String authenticate() {
+    @Override
+    public String toSubscribeMessage() {
```

Comment:
Just moved to be first as public (high level) method, the private (low level, detail) methods come after it. Also fixed warnings too about redundant generics at the definition side.